include/OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp

```diff
@@ -164,63 +161,80 @@ class NumericalSolver
+    Array<Solution> integrateTimes(
         const StateVector& anInitialStateVector,
-        const Instant& aStartInstant,
-        const Array<Instant>& anInstantArray,
+        const Real& aStartTime,
+        const Array<Real>& aTimeArray,
         const SystemOfEquationsWrapper& aSystemOfEquations
     );
 
-    /// @brief              Perform numerical integration from an instant to another instant
+    /// @brief              Perform numerical integration from a start time to an end time
     ///
     /// @code
-    ///                     StateVector stateVector = numericalSolver.integrateStateFromInstantToInstant(stateVector,
-    ///                     instant, otherInstant, systemOfEquations) ;
+    ///                     Solution solution = numericalSolver.integrateTimes(stateVector, startTime, endTime,
+    ///                     systemOfEquations);
     /// @endcode
     /// @param              [in] anInitialStateVector An initial n-dimensional state vector to begin integrating at
-    /// @param              [in] aStartInstant An instant to begin integrating from
-    /// @param              [in] anEndInstant An instant to finish integrating at
-    /// @param              [in] aSystemOfEquations An std::function wrapper with a particular signature that
+    /// @param              [in] aStartTime A time to begin integrating from
+    /// @param              [in] anEndTime An time to integrate to
+    /// @param              [in] aSystemOfEquations A std::function wrapper with a particular signature that
     /// boost::odeint accepts to perform numerical integration
     /// @return             std::vector<double>
 
-    StateVector integrateStateFromInstantToInstant(
+    Solution integrateTimes(
         const StateVector& anInitialStateVector,
-        const Instant& aStartInstant,
-        const Instant& anEndInstant,
+        const Real& aStartTime,
+        const Real& anEndTime,
         const SystemOfEquationsWrapper& aSystemOfEquations
     );
 
     /// @brief              Perform numerical integration for a certain duration
     ///
     /// @code
-    ///                     StateVector stateVector = numericalsolver.integrateStateFromInstantToInstant(stateVector,
-    ///                     instant, otherInstant, SystemofEquations) ;
+    ///                     Solution solution = numericalsolver.integrateTimes(stateVector, durationSeconds,
+    ///                     SystemofEquations) ;
+    /// @endcode
+    /// @param              [in] anInitialStateVector An initial n-dimensional state vector to begin integrating at
+    /// @param              [in] aDurationInSeconds A duration over which to integrate
+    /// @param              [in] aSystemOfEquations A std::function wrapper with a particular signature that
+    /// boost::odeint accepts to perform numerical integration
+    /// @return             std::vector<double>
+
+    Solution integrateDurations(
```

Comment:
I just wanted a single overloaded function so that we don't have to check in python `if len(array) == 0 -> integrateDuration else integrateDurations`.